### PR TITLE
add linkerd run benchmark job script

### PR DESCRIPTION
**Pull request by carolynhu**
In this PR:
1. reorganized benchmark config files: `configs/istio` and `configs/linkerd`
2. renamed `run_benchmark_job.sh` to `run_istio_benchmark_job.sh`
3. add `run_linkerd_benchmark_job.sh`.

Note: currently, we have not supported collecting linkerd data from `svc/linkerd-prometheus`, which will be added in later PRs. So, in this PR, I only dumped fortio rawdata from `fortioclient` pod to `LOCAL_OUTPUT_DIR`.

### CODE REVIEW

File: perf/benchmark/run_linkerd_benchmark_job.sh [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+
+# Exit immediately for non zero status
+set -e
+# Check unset variables
+set -u
+# Print commands
+set -x
+
+# This is config for postsubmit cluster.
+export VALUES="${VALUES:-values-istio-postsubmit.yaml}"  # TODO: how to set this for linkerd or we do not need this for linkerd?
```
Comment (carolynhu) [inline perf/benchmark/run_linkerd_benchmark_job.sh]: @richardwxn what do you think?

**Comment by carolynhu** [comment]
@richardwxn linkerd_version can be configured. What we need to do is to always tracking the latest linked release when the current hard-coded one outdated.

**Comment by richardwxn** [comment]
@carolynhu yep what I mean is that we may need to figure our the linkerd release process(for example do they have a dev builds or daily releases or sth different) and set it up in this PR, and configs the jobs correspondingly

**Comment by howardjohn** [comment]
@richardwxn makes a good point. Should we just run this manually when linkerd creates a release?

**Comment by carolynhu** [comment]
@howardjohn I think that would be the easiest way, since we do not need to figure out how linkerd process their release cycles. Then, we are no longer need to create a periodic job anymore?

**Comment by richardwxn** [comment]
@carolynhu yep we may not need a periodic job.

**Comment by carolynhu** [comment]
@richardwxn @howardjohn OK, then, we can create a script to run linkerd perf tests in the `configs/linkerd` repo. And run that script manually when linkerd has a new release.

**Comment by howardjohn** [comment]
@carolynhu sounds good to me!

File: perf/benchmark/run_linkerd_benchmark_job.sh (hunk 1 above)
Comment (mandarjog) [reply perf/benchmark/run_linkerd_benchmark_job.sh]: We need to make sure that we are providing appropriate CPU / memory.
@richardwxn are these values anything other than cpu/mem limits?

**Comment by wmorgan** [comment]
Where will these results be published?

**Comment by mandarjog** [comment]
@wmorgan at present they will be placed in a gcs bucket.
This is so that we know where latency measurements are.
I want to ensure that linkerd is configured appropriately here, to that end can you review the linkerd config ? Thanks.

**Comment by mandarjog** [comment]
At some point I would like meshery to run this benchmark as well. @leecalcote

**Comment by wmorgan** [comment]
A bucket that only the Istio team has access to?

**Comment by mandarjog** [comment]
All results are placed in public buckets.

**Comment by leecalcote** [comment]
Meshery has recently incorporated support for both `fortio` and `wrk2` and are finishing off the project's ability to drive performance tests headlessly. We're committed to seeing Meshery incorporated into Istio and Linkerd (and other meshes') CI / release / performance testing processes such that a consistent set of tests, configurations and result formats are used.

Screen Shot 2019-12-09 at 6 50 19 PM

Along with a group focused on VM-based workload profile definition and measurement, we're spinning up a new LF project to land, discuss, and advance these common specifications. All of your participation is very much desired (details and invitation to participate forthcoming).

**Pull request merged**